ludwig/explain/captum.py

```diff
@@ -158,51 +176,132 @@ def explain(self) -> Tuple[List[Explanation], List[float]]:
-                explanation.add(le_true.feature_attributions * -1)
+                negated_attributions = le_true.to_array() * -1
+                negated_token_attributions = {
+                    fa.feature_name: [(t, -a) for t, a in fa.token_attributions]
+                    for fa in le_true.feature_attributions
+                    if fa.token_attributions is not None
+                }
+                explanation.add(input_features.keys(), negated_attributions, negated_token_attributions)
 
             # TODO(travis): for force plots, need something similar to SHAP E[X]
             expected_values.append(0.0)
 
         return self.explanations, expected_values
 
 
-def get_baseline(sample_encoded: List[Variable]) -> List[Variable]:
-    # For a robust baseline, we take the mean of all embeddings in the sample from the training data.
+def get_baseline(model: LudwigModel, sample_encoded: List[Variable]) -> List[torch.Tensor]:
     # TODO(travis): pre-compute this during training from the full training dataset.
-    return [torch.unsqueeze(torch.mean(t, dim=0), 0) for t in sample_encoded]
+    input_features: LudwigFeatureDict = model.model.input_features
+
+    baselines = []
+    for sample_input, (name, feature) in zip(sample_encoded, input_features.items()):
+        metadata = model.training_set_metadata[name]
+        if feature.type() == TEXT:
+            PAD_IND = metadata["pad_idx"]
+            token_reference = TokenReferenceBase(reference_token_idx=PAD_IND)
+            baseline = token_reference.generate_reference(sequence_length=sample_input.shape[1], device=DEVICE)
+        elif feature.type() == CATEGORY:
+            most_popular_token = max(metadata["str2freq"], key=metadata["str2freq"].get)
+            most_popular_tok_idx = metadata["str2idx"].get(most_popular_token)
+
+            # If an unknown is defined, use that as the baseline index, else use the most popular token
+            baseline_tok_idx = metadata["str2idx"].get(UNKNOWN_SYMBOL, most_popular_tok_idx)
+            baseline = torch.tensor(baseline_tok_idx, device=DEVICE)
+        else:
+            # For a robust baseline, we take the mean of all embeddings in the sample from the training data.
+            # TODO(joppe): now that we don't have embeddings, we should re-evaluate this.
+            baseline = torch.mean(sample_input.float(), dim=0)
+        baselines.append(baseline.unsqueeze(0))
+
+    return baselines
 
 
 def get_total_attribution(
     model: LudwigModel,
     target_feature_name: str,
     target_idx: Optional[int],
-    inputs_encoded: List[Variable],
-    baseline: List[Variable],
+    feature_inputs: List[Variable],
+    baseline: List[torch.Tensor],
     use_global: bool,
     nsamples: int,
-) -> np.array:
+) -> Tuple[npt.NDArray[np.float64], Dict[str, List[List[Tuple[str, float]]]]]:
+    """Compute the total attribution for each input feature for each row in the input data.
```

Comment:
nit: Use Google python style docstrings:
https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
Sounds good, will change now!